esmvalcore/_recipe.py

```diff
@@ -1061,6 +1062,12 @@ def expand_ensemble(variable):
 
         return expanded
 
+    def _add_project_variable_mappings(self, variable):
+        mappings = get_variable_mappings(variable["project"],
+                                         variable["dataset"])
+        mapping = mappings[variable["mip"]][variable["short_name"]]
```

Comment:
"mip" shouldn't be a key in mapping : the actual need for mapping is to describe various alternate variable names, and maybe also scaling/offset attributes; these attributes depend on the base variable name, but not depend  on the CMIP table name, which depend on data frequency; and there is no ambiguity of base variable name across tables (except maybe for one single variable in CMIP6)

Reply:
Hm. I am not sure about this. For example in era5 data, we might want to derive daily means from hourly data and monthly means from, well, monthly data. Those live in different files and thus we need a different mapping. For models, the needs may be different, but first, the overhead is relatively low because the same variable appears only in a handful of tables, and second, it may still sometimes be useful. For example, occasionally there is both an instantaneous version of, say `pr`, and a time integrated one. One should  be used for tables like `3hrPt`, the other for tables like `3hr`.